index.js

```diff
@@ -1,47 +1,48 @@
 'use strict';
 const path = require('path');
-const loadJsonFile = require('load-json-file');
-const pathType = require('path-type');
+const fs = require('graceful-fs');
+const normalizePackageData = require('normalize-package-data');
 
-module.exports = (fp, opts) => {
-	if (typeof fp !== 'string') {
-		opts = fp;
-		fp = '.';
-	}
-
-	opts = opts || {};
+const PACKAGE_FILE = 'package.json';
 
-	return pathType.dir(fp)
-		.then(isDir => {
-			if (isDir) {
-				fp = path.join(fp, 'package.json');
+function readJsonFileAsync(file) {
+	return new Promise((resolve, reject) => {
+		fs.readFile(file, 'utf8', (err, data) => {
+			if (err) {
+				return reject(err);
 			}
-
-			return loadJsonFile(fp);
-		})
-		.then(x => {
-			if (opts.normalize !== false) {
-				require('normalize-package-data')(x);
-			}
-
-			return x;
+			resolve(data);
 		});
-};
+	});
+}
+
+function packagePath(file = path.resolve()) {
+	if (path.extname(file)) {
```

Comment:
This is not a reliable way to check if something is a file or directory. Directories can have `.` in them too.

Reply:
Is this worth the extra FS call, though? Shouldn't this be the up to the user?

Reply:
Remember, this is used to read lots of packages. Is 100 package file reads worth 200 fs calls?

Reply:
It's a cheap call, but I think the API could be improved regardless. Let's remove support for specifying a full path and only support specifying a directory. Instead of `readPkg(__dirname)`, we can make it `readPkg({cwd: __dirname})`. Now we don't have to do the check.

Reply:
@sindresorhus how about `file.endsWith('package.json')`?